### PR TITLE
Add wireguard to xray outbound

### PR DESCRIPTION
**Pull request by sudospaes**
Xray (1.6.5+) has added outbound WireGuard support. 
With this feature, we can use Warp or any WireGuard configuration in outbounds. I added the WireGuard template to x-ui to make it easier to use this feature.

Screenshot 2023-12-14 130400

Screenshot 2023-12-14 130415

Thank you for developing x-ui alireza ❤️

### CODE REVIEW

File: web/html/xui/form/outbound.html [hunk 1]
```diff
@@ -66,6 +66,31 @@
         </a-form-item>
 </template>
 
+<!-- wireguard settings -->
+<template v-if="outbound.protocol === Protocols.Wireguard">
+    <a-form-item label='{{ i18n "pages.xray.outbound.MTU" }}'>
+        <a-input-number v-model.number="outbound.settings.MTU"></a-input-number>
+    </a-form-item>
+    <a-form-item label='{{ i18n "pages.xray.outbound.DNS" }}'>
```
Comment (alireza0) [inline web/html/xui/form/outbound.html:74]: Please don't use i18n translation for well-known abbreviations.
DNS and MTU should be used as same as english words.

File: web/translation/translate.en_US.toml [hunk 2]
```diff
@@ -388,6 +388,12 @@
 "bridge" = "Bridge"
 "portal" = "Portal"
 "intercon" = "Interconnection"
+"MTU" = "MTU"
+"DNS" = "DNS"
```
Comment (alireza0) [inline web/translation/translate.en_US.toml:392]: As above

File: web/translation/translate.fa_IR.toml [hunk 3]
```diff
@@ -387,6 +387,12 @@
 "bridge" = "پل"
 "portal" = "پرتال"
 "intercon" = "اتصال میانی"
+"MTU" = "MTU"
+"DNS" = "DNS"
```
Comment (alireza0) [inline web/translation/translate.fa_IR.toml:391]: Same

File: web/translation/translate.ru_RU.toml [hunk 4]
```diff
@@ -388,6 +388,12 @@
 "bridge" = "Мост"
 "portal" = "Портал"
 "intercon" = "Соединение"
+"MTU" = "МТУ"
+"DNS" = "DNS"
```
Comment (alireza0) [inline web/translation/translate.ru_RU.toml:392]: Same

File: web/translation/translate.vi_VN.toml [hunk 5]
```diff
@@ -388,6 +388,12 @@
 "bridge" = "Liên kết"
 "portal" = "Cổng thông tin"
 "intercon" = "Kết nối"
+"MTU" = "MTU"
+"DNS" = "DNS"
```
Comment (alireza0) [inline web/translation/translate.vi_VN.toml:392]: Same

File: web/translation/translate.zh_Hans.toml [hunk 6]
```diff
@@ -388,6 +388,12 @@
 "bridge" = "桥"
 "portal" = "门户"
 "intercon" = "互连"
+"MTU" = "最大传输单元"
+"DNS" = "域名系统"
```
Comment (alireza0) [inline web/translation/translate.zh_Hans.toml:392]: Same

**Review by alireza0 — CHANGES_REQUESTED** [review]
Thank you for your enhancements.
Actually, I wait for new update's document from xray-core which they support this protocol in inbounds from early version.
It is complete and clean! Good job!

Just please review the comments regarding using i18n and update this PR.
Thanks again

File: web/assets/js/model/outbound.js [hunk 7]
```diff
@@ -897,4 +900,43 @@ Outbound.HttpSettings = class extends CommonClass {
             }],
         };
     }
+};
+Outbound.WireguardSettings = class extends CommonClass{
+    constructor(MTU, DNS, secretKey, address, publicKey, allowedIPs, endpoint) {
```
Comment (alireza0) [inline web/assets/js/model/outbound.js:905]: It is also better to pre-define constructor values as much as possible, for example MTU and DNS

Comment (alireza0) [reply web/assets/js/model/outbound.js:905]: Also use small letters for parameter's name: mtu, dns

**Comment by sudospaes** [comment]
The requested changes have been applied https://github.com/alireza0/x-ui/pull/763/commits/622772eb6fa36996d358e03de5e1e7f5f23f9b9c

File: web/assets/js/model/outbound.js [hunk 8]
```diff
@@ -897,4 +900,43 @@ Outbound.HttpSettings = class extends CommonClass {
...
+            peers[0].publicKey,
+            peers[0].allowedIPs.toString(),
+            peers[0].endpoint,
+        );
+    }
+
+    toJson() {
+        return {
+            MTU: this.mtu,
+            DNS: this.dns,
```
Comment (alireza0) [inline web/assets/js/model/outbound.js:932]: mtu should be defined here in small leters

**Review by alireza0 — CHANGES_REQUESTED** [review]
As I checked, there is no `dns` parameter is `Wireguard` outbound streamSetting.
[please check here](https://xtls.github.io/en/config/outbounds/wireguard.html)

And also please use small letters for `mtu` in toJson() method

**Comment by sudospaes** [comment]
> As I checked, there is no `dns` parameter in `Wireguard` outbound streamSetting. [please check here](https://xtls.github.io/en/config/outbounds/wireguard.html)
> 
> And also please use small letters for `mtu` in toJson() method

I have seen it elsewhere. DNS is even available in the Wireguard interface and its clients.

**Comment by sudospaes** [comment]
I removed "DNS" field because it is not in the xray-core document. Also "keepAlive" field has been added to the template. https://github.com/alireza0/x-ui/pull/763/commits/47949a7c0d50205859feba36a664c2bf68e3a0a1

**Comment by alireza0** [comment]
Dear @sudospaes,

Thank you for you effort.

There are some bugs in xray 1.8.6 related to the `wireguard` protocol. I will keep this request be opened till all changes in xray-core be fixed.

**Comment by sudospaes** [comment]
> Dear @sudospaes,
> 
> Thank you for you effort.
> 
> There are some bugs in xray 1.8.6 related to the `wireguard` protocol. I will keep this request be opened till all changes in xray-core be fixed.

Thank you @alireza0

**Comment by kalilovers** [comment]
how we can use other wireguard (no warp) ? do we had to make a personal wireguard server for use in our panel or there is other freeone like warp?

**Review by alireza0 — APPROVED** [review]

**Comment by alireza0** [comment]
@kalilovers Inbounds and Outbound part are doing different activities.
Outbound in one server will connect to an Inbound from another server.

**Comment by kalilovers** [comment]
> @kalilovers Inbounds and Outbound part are doing different activities.
> Outbound in one server will connect to an Inbound from another server.

Understood, this is very helpful!

Look, I'm right, if I create a wireguard-type outbound, in this case, I create an inbound, for example, on the American server, then I create the outbound on the my main server, then I can create a rules consisting of several sites and place  Output it to Wireguard Outband, transfer that list from my US server, a bit like Warp, right?

**Pull request merged**